assets/js/components/PresenterStreams.tsx

```diff
@@ -18,25 +25,20 @@ const playerCallbacks: { [key: string]: (sourceType: SourceType) => void } = {};
 let webrtc: MembraneWebRTC | null = null;
 let webrtcConnecting: boolean = false;
 
-const PresenterStreams = ({
-  clientName,
-  eventChannel,
-  mode,
-  setMode,
-}: PresenterStreamAreaProps) => {
-  const [presenters, setPresenters] = useState<string[]>([]);
+const PresenterStreams = ({ client, eventChannel, mode, setMode }: PresenterStreamAreaProps) => {
+  const [presenters, setPresenters] = useState<Presenter[]>([]);
   const [isControlPanelAvailable, setIsControlPanelAvailable] = useState(false);
 
   useEffect(() => {
-    if (!webrtcConnecting && webrtc == null && presenters.includes(clientName)) {
+    if (!webrtcConnecting && webrtc == null && presenters.some((e) => e.email === client.email)) {
```

Comment:
We use `presenters.some((e) => e.email == client.email` in both conditions. Move it to variable.